src/Protocol/Authenticator.cpp

```diff
@@ -51,6 +50,24 @@ void cAuthenticator::ReadSettings(cSettingsRepositoryInterface & a_Settings)
 	m_Server             = a_Settings.GetValueSet ("Authentication", "Server", DEFAULT_AUTH_SERVER);
 	m_Address            = a_Settings.GetValueSet ("Authentication", "Address", DEFAULT_AUTH_ADDRESS);
 	m_ShouldAuthenticate = a_Settings.GetValueSetB("Authentication", "Authenticate", true);
+
+	// prepend https:// if missing
+	if (m_Server.rfind("http") == AString::npos)
+	{
+		m_Server = "https://" + m_Server;
+	}
```

Comment:
I still think `rfind` is not the right one. We want "If it doesn't start with `http://` or `https://`, then prepend `https://`"